src/grpc/headers-interceptor.ts

```diff
@@ -36,10 +37,12 @@ export class HeaderInterceptor {
     return (options, nextCall) => {
       return new InterceptingCall(nextCall(options), {
         start: (metadata, listener, next) => {
+          metadata.add('request-id', randomUUID());
```

Comment:
we should talk about this.  I add this because it was useful in helping me debug some things related to retries (otherwise it is very difficult to distinguish requests from one another).  I also assume that it will be generally useful for server-side traces, and possibly for client-side traces/metrics.  But it does add some bytes to the payload, so happy to discuss, and possibly remove / defer to a later time.

Reply:
my 2 cents is its 100% worth the extra bytes having this on every request will make operations much easier prevent us wasting time trouble shooting issues especially now that we are starting to do client side retrys.

Would rather start with this on by default and wait for customer to have issue with bandwidth and then provide an optional way to toggle this header off for very advanced high load users.

I feel similarly about user agent header being sent on every request. Would rather we default to easier operations and reasonable performance and come back and solve for the extreme case when needed and provide ways to toggle off for advanced users.

Reply:
If we are not tracking this at the service layer I am not sure how we will use this? Is it for client side debugging?

Reply:
I am 💯 on board for request ids

Reply:
I'm onboard with adding this but it might be helpful to cross check this with OTel tracing specs (not metrics, don't hate, their tracing protocol is pretty good :P) to ensure we use the standard header tag. There is grpc middleware that automatically links up your traces if you use standard headers (we probably don't have that middleware wrapped around the service yet but we could add it easily later). I can help with this later today if needed.

Reply:
I'm pretty sure the grpc middleware we have on mr2 already does this @danielamiao 

Agree otlp tracing is not bad at all (otel tracing library though...)

I'm not a fan of doing this for every request. We're a cache, not a database. We should support this but have it default to off - it's an extra 32 bytes per request of overhead paid for by users. Also, it's unnecessary tracking data which we should be cautious with and mindful of.

Reply:
> We should support this but have it default to off 

We seem to keep running into these types of issues that require us to go to the customer to tell them to do something so we can debug. I would prefer if we could debug customer requests without having to ask the customer to help us.

Reply:
If we are generally okay with the idea of this existing, but possibly being configurable, we can defer the decision about how to configure it and what the defaults should be until https://github.com/momentohq/dev-eco-issue-tracker/issues/85 , which I intend to pick up very soon.

I don't want to make it configurable in this PR.  For this PR I think the options are either always-on, or remove this code for now.

I can double-check against the otel specs for header name once we have agreed whether or not to even include this in this PR.

Reply:
And you're going to do that with the client's uuid how matt?

It's extra cost footed by the user per request, bloating 100% of requests with tracking information that is extremely debatably useful for 0.000001% of requests.. and again, how are you going to do anything with it without asking the user for something? It's logged client side at debug. If debug logging isn't enabled, then it's completely wasted network transfer - a client side id generator for our service to.. idk what actually

Also, fyi, we do head sampling of request traces. There is a real possibility that uuids a user asks you about will have no trace on an active system and you won't be able to disambiguate "your request never reached us" from "I didn't trace that request"

What is the value offered to users that is worth the cost of 32 bytes additional per request overhead to their cache? Happy to take this to slack

Reply:
@cprice404 let's remove it from this PR to unblock this PR for now. That way we can agree on defaults + use the otlp tracing header in the follow up PR in one go.

Reply:
@danielamiao agreed, I will do that.

Reply:
removed in https://github.com/momentohq/client-sdk-javascript/pull/84/commits/912471012bc256cf275766de7e83ad7e06efd1e9